features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleHistory.jsx

```diff
@@ -64,7 +64,8 @@ const LifeCycleHistory = (props) => {
                     </TableRow>
                 </TableHead>
                 <TableBody>
-                    {props.lcHistory.map((entry) => entry.previousState && (
+                    {props.lcHistory.sort((a, b) => moment(b.updatedTime)
```

Comment:
@Sarangan0219 Shall we use `Array.prototype.reverse()` the method here, I think we get the lc history array in reverse order and re-reversing it before rendering would fix this issue

Reply:
it is less time complex compared to sorting https://github.com/trekhleb/javascript-algorithms#Sorting

https://www.geeksforgeeks.org/time-complexities-of-all-sorting-algorithms/

😃

Reply:
https://github.com/wso2/carbon-apimgt/pull/8243/commits/3815fdcd38e73d1a361c8d4b830fe0e267794d7d

![Screenshot from 2020-02-26 12-08-21](https://user-images.githubusercontent.com/60866295/75318817-5521c480-5891-11ea-867b-ea0b90142f23.png)

We don't need to re reverse before rendering. lchistory is behaving like a queue in getting the inputs .Thus reversing at the start and rendering is enough to get the desired behaviour.